cpplint.py

```diff
@@ -979,26 +1006,38 @@ def ParseNolintSuppressions(filename, raw_line, linenum, error):
     linenum: int, the number of the current line.
     error: function, an error handler.
   """
-  matched = Search(r'\bNOLINT(NEXTLINE)?\b(\([^)]+\))?', raw_line)
+  matched = Search(r'\bNOLINT(NEXTLINE|BEGIN|END)?\b(\([^)]+\))?', raw_line)
   if matched:
-    if matched.group(1):
-      suppressed_line = linenum + 1
+    no_lint_type = matched.group(1)
+    if no_lint_type == 'NEXTLINE':
+      process_category = lambda c: _error_suppressions.setdefault(c, set()).add(linenum + 1)
+    elif no_lint_type == 'BEGIN':
+      process_category = lambda c: _block_error_suppressions.update((c,))
```

Comment:
So I am not sure why there should ever be a reason to go beyond the value of 1.

Reply:
This was done to handle the possibility of the same category being ignored multiple times in nested blocks but I will make that an error instead.